semgrep-core/src/parsing/tree_sitter/Parse_dockerfile_tree_sitter.ml

```diff
@@ -307,6 +309,7 @@ let unquoted_string (env : env) (xs : CST.unquoted_string) : str =
         | `BSLASHSPACE tok -> String_content (str env tok (* "\\ " *))
         | `Imme_expa x -> expansion env x)
       xs
+    |> simplify_fragments
```

Comment:
I wonder if the type for fragments could be changed to enforce the invariant that you should call simplify_fragments.
(a bit like what Brandon did recently with his new Path.ml module).

Reply:
We could have a new type with exactly the same structure but which would have to be created with `simplify_fragments`. I think it's a bit overkill but I'll try it as an exercise.

Reply:
I [tried using a private type abbreviation](https://github.com/returntocorp/semgrep/pull/6144/files#r975918900) to ensure that the type `str` has no fragmented literals but it poses problems with pattern matching.